src/js/game/hud/parts/waypoints.js

```diff
@@ -228,6 +231,7 @@ export class HUDWaypoints extends BaseHUDPart {
      * @param {Waypoint} waypoint
      */
     moveToWaypoint(waypoint) {
+        this.root.currentLayer = waypoint.layer;

```

Comment:
You'll have to migrate old savegames, otherwise it will break here since layer is null then

Reply:
oh oops i completely forgot! I'll fix it!